hotsos/plugin_extensions/openstack/agent/events.py

```diff
@@ -241,14 +247,19 @@ def __call__(self, event):
                 results.append({'date': e.get(1), 'time': e.get(2),
                                 'key': lb_id})
 
-            ret = self.categorise_events(event, results=results,
-                                         key_by_date=False)
+            ret = self.categorise_events(
+                event,
+                results=results,
+                options=type(self).EventProcessingOptions(key_by_date=False)
```

Comment:
nit: it isn't necessary to use type() here? 
 since this does work either why i wont block on this and can fix in followup patch

Reply:
It's not necessary, it's just shorter to type `type(self).` instead of `EventProcessingUtils.`

Reply:
Also, I wanted to avoid using `self.` because it looks like a method call to me, which is not the case. But it might still be confused with a static or class method call, so I doubt the merit of doing that now.

Reply:
this is why i dont like nested functions and classes ;)

Reply:
well, that and they are harder to test